src/sdk-reference/cpp/1/getting-started/snippets/realtime.cpp

```diff
@@ -1,72 +1,79 @@
+  // Replace "kuzzle" with your actual Kuzzle hostname (e.g. "localhost")
+  kuzzleio::WebSocket *ws = new kuzzleio::WebSocket("kuzzle");
+  kuzzleio::Kuzzle *kuzzle = new kuzzleio::Kuzzle(ws);
 
-    // Starts an async listener
-    kuzzleio::NotificationListener listener =
-      [](const std::shared_ptr<kuzzleio::notification_result> &notification) {
-        const char *id = notification->result->id;
-        std::cout << "New created document notification: " << id << std::endl;
-      };
+  try {
+    // Connect to the server.
+    kuzzle->connect();
+    std::cout << "Connected!" << std::endl;
+  }
+  catch(kuzzleio::KuzzleException &e) {
+    std::cerr << e.what() << std::endl;
+    exit(1);
+  }
 
-    try {
-      // Subscribes to notifications for drivers having a "B" driver license.
-      const char *filters = R"(
-        {
-           "equals": {
-             "license": "B"
-           }
-        }
-      )";
+  // Create a lambda that will be invoked for each real-time notification
+  // received
+  kuzzleio::NotificationListener listener =
+    [](const std::shared_ptr<kuzzleio::notification_result> &notification) {
+      const char *id = notification->result->id;
+      std::cout << "New created document notification: " << id << std::endl;
+    };
 
-      // Sends the subscription
-      kuzzle->realtime->subscribe(K_INDEX_NAME, K_COLLECTION_NAME, filters, &listener);
-      std::cout << "Successfully subscribed!" << std::endl;
+  try {
+    // Subscribe to notifications for drivers having a "B" driver license.
+    const char *filters = R"(
+      {
+         "equals": {
+           "license": "B"
+         }
+      }
+    )";
 
-      // Writes a new document. This triggers a notification
-      // sent to our subscription.
-      const char *document = R"(
-        {
-          "name": "John",
-          "birthday": "1995-11-27",
-          "license": "B"
-        }
-      )";
+    kuzzle->realtime->subscribe(
+        "nyc-open-data",
+        "yellow-taxi",
+        filters,
+        &listener);
+    std::cout << "Successfully subscribed!" << std::endl;
 
-      kuzzle->document->create(K_INDEX_NAME, K_COLLECTION_NAME, "some-id", document);
-      // Wait for notification
-      sleep(2);
-    } catch (kuzzleio::KuzzleException &e) {
-      std::cerr << e.what() << std::endl;
-      kuzzle->disconnect();
-      exit(1);
-    }
+    // Write a new document. This triggers a notification
+    // sent to our subscription.
+    const char *document = R"(
+      {
+        "name": "John",
+        "birthday": "1995-11-27",
+        "license": "B"
+      }
+    )";
 
-    // Disconnects the SDK.
+    kuzzle->document->create(
+        "nyc-open-data",
+        "yellow-taxi",
+        "some-id",
+        document);
+  } catch (kuzzleio::KuzzleException &e) {
+    std::cerr << e.what() << std::endl;
     kuzzle->disconnect();
+    exit(1);
+  }
+
+  // Wait for the notification to be received
+  sleep(2);
```

Comment:
You should use a retry mechanism instead to avoid failing test when Travis is on heavy load.
I have already done that in this template: https://github.com/kuzzleio/documentation/blob/master/test/templates/realtime.tpl.cpp

Reply:
Fixed.